awx/api/fields.py

```diff
@@ -101,9 +101,15 @@ def __init__(self, **kwargs):
         super(DeprecatedCredentialField, self).__init__(**kwargs)
 
     def to_internal_value(self, pk):
+        if isinstance(pk, str):
+            self.fail('invalid')
```

Comment:
I don't think you need this, because as you pointed out, other fields will accept strings if they can be converted into integers.